examples/gazebo_differential_drive_plugin/differential_drive_plugin.cc

```diff
@@ -0,0 +1,110 @@
+#ifndef _DIFFERENTIAL_DRIVE_PLUGIN_HH_
+#define _DIFFERENTIAL_DRIVE_PLUGIN_HH_
+
+#include <gazebo/gazebo.hh>
+#include <gazebo/physics/physics.hh>
+#include <gazebo/transport/transport.hh>
+#include <gazebo/msgs/msgs.hh>
+
+namespace gazebo
+{
+  /// \brief A plugin to control a simple cart with differential drive.
+  class DifferentialDrivePlugin : public ModelPlugin
+  {
+    /// \brief Constructor
+    public: DifferentialDrivePlugin() {}
+
+    /// \brief The load function is called by Gazebo when the plugin is
+    /// inserted into simulation
+    /// \param[in] _model A pointer to the model that this plugin is
+    /// attached to.
+    /// \param[in] _sdf A pointer to the plugin's SDF element.
+    public: virtual void Load(physics::ModelPtr _model, sdf::ElementPtr _sdf)
```

Comment:
In C++ you don't need to mark every function as ``public``, ``private`` etc. (cf. e.g. Java). You usually just declare them all in blocks:
```C++
class MyClass {
public:
    void method1() { ... }
    void method2() { ... }

private:
    std::string m_SomeMember;
    void method3() { ... }
};
```